tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -90,6 +96,11 @@ syntaxDatatype language allSymbols datatype = skipDefined $ do
     generatedDatatype name cons typeParameterName = DataD [] name [PlainTV typeParameterName] Nothing cons [deriveStockClause, deriveAnyClassClause]
 
 
+makeHasFieldInstance :: TypeQ -> TypeQ -> ExpQ -> Q [Dec]
+makeHasFieldInstance ty param elim =
+  [d|instance HasField "ann" $(ty `appT` param) $param where
+      getField = TS.gann . $elim |]
```

Comment:
This instance means that we can use `getField "ann"` to access the annotation of all our AST types, e.g. this bit where I’m working against the `tree-sitter-json` AST:

```
λ getField @"ann" (Value (L1 (R1 (R1 (AST.Null (100) "")))))
100
```

we can further use `gann` against anonymous sums; I had considered wrapping anonymous sums in a `Sum` newtype that would also provide a `HasField "ann"` instance, but I decided it’d be too intrusive a change for this PR.

Reply:
A little unorthodox, but there’s nothing stopping us from doing it, and the documents specifically say that this is allowed, so 👍 .